kahypar/partition/metrics.h

```diff
@@ -87,7 +87,7 @@ static inline HyperedgeWeight soed(const Hypergraph& hg) {
 static inline HyperedgeWeight km1(const Hypergraph& hg) {
   HyperedgeWeight k_minus_1 = 0;
   for (const HyperedgeID& he : hg.edges()) {
-    k_minus_1 += (hg.connectivity(he) - 1) * hg.edgeWeight(he);
+    k_minus_1 += std::max(hg.connectivity(he) - 1, 0) * hg.edgeWeight(he);
```

Comment:
Do you need the function during initial partitionining? Otherwise I don't understand why the max is necessary since conenctivity should always be at least 1 after IP.

Reply:
Not in RB mode (fixed vertices are not assigned before maximum bipartite weighted matching), if fixed vertices are not assigned and a hyperedge consists completly of fixed vertices, than the connectivity will be zero and k_minus_1 = - hg.edgeWeight(he) ;)